src/digital_identity_service/index.js

```diff
@@ -9,6 +13,45 @@ class DigitalIdentityService {
     this.pem = pem;
     this.apiUrl = apiUrl;
   }
+
+  /**
+   *
+   * @param shareSessionConfig
+   * @returns {Promise<ShareSessionResult>}
+   */
+  createSession(shareSessionConfig) {
+    const jsonShareSession = JSON.parse(JSON.stringify(shareSessionConfig));
+
+    const payload = new Payload(jsonShareSession);
+
+    const requestBuilder = new RequestBuilder()
+      .withBaseUrl(this.apiUrl)
+      .withHeader('X-Yoti-Auth-Id', this.sdkId)
+      .withPemString(this.pem)
+      .withEndpoint('/v2/sessions')
+      .withQueryParam('appId', this.sdkId)
+      .withMethod('POST')
+      .withPayload(payload);
+
+    const request = requestBuilder.build();
+
+    return new Promise((resolve, reject) => {
```

Comment:
Maybe its personal opinion but I don't see a good reason for not having these class methods as `async` and handling errors in `try` `catch`. Maybe there is a reason for this though.

Reply:
I do agree with you, so yes maybe in this new service we should follow the pattern of `async` with `try` `catch`.
What do you think @laurent-yoti , @michaelyoti , @liamd-yoti ?

Reply:
Consistency wise, it is easier to keep one style... and then change them all once we move to TS.
Yet, we can already considering "educating" users with `async/await` now.